src/FiltrationConditionComponent/FiltrationConditionComponent.tsx

```diff
@@ -9,15 +9,62 @@ import "./FiltrationConditionComponent.scss";
 
 type Props = StateProps & CallbackProps;
 
-export class FiltrationConditionComponent extends React.Component<Props> {
+interface State {
+	offsetTop: number;
+}
+
+const RANGE_OFFSET_TOP = 30;
+
+export class FiltrationConditionComponent extends React.Component<
+	Props,
+	State
+> {
 	public refComponent = React.createRef<HTMLElement>();
 	public refContent = React.createRef<HTMLDivElement>();
 
+	public state = {
+		offsetTop: 0
+	};
+
 	public componentDidMount() {
-		this.handleMoveUpParentPopupSergment();
+		window.addEventListener("scroll", this.scrollWindowsHeight);
+		this.handleMoveUpParentPopupSegment();
 	}
 
-	public handleMoveUpParentPopupSergment = () => {
+	public componentDidUpdate() {
+		window.addEventListener("scroll", this.scrollWindowsHeight);
+	}
+
+	public componentWillUnmount() {
+		window.removeEventListener("scroll", this.scrollWindowsHeight);
+	}
+
+	public scrollWindowsHeight = () => {
+		const { offsetTop } = this.state;
+		const { onConditionStateToggle, state } = this.props;
+		const refContent = this.refContent.current;
+
+		if (refContent) {
+			const rect = refContent.getBoundingClientRect();
```

Comment:
Можно немного убрать мельтишение использовав `top` вместо `rect.top`
```suggestion
			const { top } = refContent.getBoundingClientRect();
```